### PR TITLE
Extend transitive availability checking to initial value expressions

### PR DESCRIPTION
**Pull request by jrose-apple**
Because initial value expressions aren't actually considered *within* the VarDecl or PatternBindingDecl they're initializing, the existing logic to search for availability attributes wasn't kicking in, leading to errors when a conditionally-unavailable value was used in an initial value expression for a conditionally-unavailable binding.

[SR-9867](https://bugs.swift.org/browse/SR-9867) / rdar://problem/47852718

### CODE REVIEW

**Comment by jrose-apple** [comment]
@swift-ci Please test

**Comment by jrose-apple** [comment]
@swift-ci Please test source compatibility

**Review by slavapestov — APPROVED** [review]

File: test/attr/attr_availability_transitive_osx.swift [hunk 1]
```diff
@@ -35,3 +39,43 @@ func osx_extension_call_osx_extension() {
 func osx_extension_call_osx() {
     osx() // expected-error {{'osx()' is unavailable}}
 }
+
+@available(OSX, unavailable)
+var osx_init_osx = osx() // OK
+
+@available(OSXApplicationExtension, unavailable)
+var osx_extension_init_osx = osx() // expected-error {{'osx()' is unavailable}}
```
Comment (brentdax) [inline test/attr/attr_availability_transitive_osx.swift]: Would this test pass?
```swift
@available(OSX, unavailable)
var osx_inner_init_osx = { let inner_var = osx() }
```
If so, is it worth adding to the test suite? If not, do we want to merge this change anyway?

**Review by beccadax — APPROVED** [review]
If the comment below doesn't prompt any changes, this looks good to me.

File: test/attr/attr_availability_transitive_osx.swift (hunk 1 above)
Comment (jrose-apple) [reply test/attr/attr_availability_transitive_osx.swift]: Definitely seems worth adding! I think it'll work with this change, but it's always worth checking.

Comment (jrose-apple) [reply test/attr/attr_availability_transitive_osx.swift]: Aha, it does *not* work. Important info!

**Comment by jrose-apple** [comment]
Found a better approach, will open a new PR for that.

**Pull request closed without merging**